### PR TITLE
add --clean-up option to remove container after run

### PR DESCRIPTION
**Pull request by m-g-k**
adding a new --clean-up option to remove the container after the run completes.

### CODE REVIEW

File: cp-deploy.sh [hunk 1]
```diff
@@ -556,6 +568,14 @@ while (( "$#" )); do
     fi
     export CPD_SKIP_PORTABLE_REGISTRY=true
     shift 1
+    ;;
+  --clean-up)
+    if [[ "${ACTION}" != "apply" && "${ACTION}" != "destroy" && "${ACTION}" != "download"  ]];then
```
Comment (fketelaars) [inline cp-deploy.sh:573]: ACTION == "vault" also qualifies for clean-up

**Review by fketelaars — CHANGES_REQUESTED** [review]
See detailed comments. Yet have to test the changes. Please send next PR to airgap-misc branch.

File: cp-deploy.sh (hunk 1 above)
Comment (m-g-k) [reply cp-deploy.sh]: Thanks for the feedback. I have added a new commit to make the `--clean-up` cmd work with the `vault` subcommand. This also meant changing `vault` to produce logs in the same was the the `environment` commands to give consistent output.

Comment (m-g-k) [reply cp-deploy.sh]: @fketelaars I've also pointed the PR to the `airgap-misc` branch as requested.

File: cp-deploy.sh [hunk 2]
```diff
@@ -922,7 +942,7 @@ if ! $INSIDE_CONTAINER;then
   fi
 
   # If running "environment" subcommand with some subcommands run as daemon
-  if [ "$SUBCOMMAND" == "environment" ] && [[ "${ACTION}" == "apply" || "${ACTION}" == "destroy" || "${ACTION}" == "wizard" || "${ACTION}" == "download" ]];then
+  if [[ "$SUBCOMMAND" == "vault" ]] || [[ "$SUBCOMMAND" == "environment" && ( "${ACTION}" == "apply" || "${ACTION}" == "destroy" || "${ACTION}" == "wizard" || "${ACTION}" == "download" ) ]]; then
```
Comment (fketelaars) [inline cp-deploy.sh:945]: The vault subcommand should not run as a daemon, please remove it from here.

File: cp-deploy.sh [hunk 3]
```diff
@@ -1039,17 +1061,23 @@ if ! $INSIDE_CONTAINER;then
   fi
   run_cmd+=" cloud-pak-deployer:${CPD_IMAGE_TAG}"
 
-  # echo $run_cmd
+   # echo $run_cmd
 
   # If running "environment" subcommand with apply/destroy, follow log
-  if [ "$SUBCOMMAND" == "environment" ] && [[ "${ACTION}" == "apply" || "${ACTION}" == "destroy" || "${ACTION}" == "wizard" || "${ACTION}" == "download" ]];then
+  if [[ "$SUBCOMMAND" == "vault" ]] || [[ "$SUBCOMMAND" == "environment" && ( "${ACTION}" == "apply" || "${ACTION}" == "destroy" || "${ACTION}" == "wizard" || "${ACTION}" == "download" ) ]]; then
```
Comment (fketelaars) [inline cp-deploy.sh:1067]: vault is run interactively, no need to follow the logs.

**Review by fketelaars — CHANGES_REQUESTED** [review]

**Pull request closed without merging**